packages/react-scripts/fixtures/kitchensink/src/App.js

```diff
@@ -1,23 +1,22 @@
-import React from 'react';
+import React, { Component, Children, cloneElement } from 'react';
 
 class BuiltEmitter extends React.Component {
-  constructor(props) {
-    super(props)
-
-    this.callWhenDone = done => done();
+  componentDidMount() {
+    const only = Children.only(this.props.children);
+    if (!Component.isPrototypeOf(only)) {
```

Comment:
The result would be React elements in both cases, so `Component` would not be their prototype.
In these examples, you'd get `{ type: 'p' | CreateP, props: {...} }`.